CONTRIBUTING.md

```diff
@@ -0,0 +1,115 @@
+# Contributing
+
+Implementing and maintaining a protocol stack is a lot of work, therefore any
+help is appreciated, from creating issues, to contributing documentation, 
+fixing issues and adding new features.
+
+Please follow the best-practice contribution guidelines as mentioned below 
+when submitting any changes.
+
+### Contributions License Agreement
+
+Before contributing, please read the 
+[Contributions License Agreement](./CLA.md).
+Every contributor must sign this agreement before any code is merged into the
+repository.
+
+### Git
+
+Please, do your changes in a dedicated feature-/fix-branch in your forked repo
+and send us the pull request from these branches in your forked repo.
+
+In the case that Commit rights to the main repository were granted to 
```

Comment:
I think on github fpor "personal account repos" (like yours right now) there are only two levels as described on https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-personal-account-settings/permission-levels-for-a-personal-account-repository

If you want to have it more granular then it would make sense to create an organization and move the repo there. Then you have some more options and granularities for all around (like issue management and such). But I think repo wise it stay on that "write or read-only".

As said: normally contributors just create a fork of the repo, update their stuff there and do the PR from there and then they pull from the main repo again. So formally no need to give out write permissions to "all" contributors (because this would also be risky these times).

Reply:
I adjusted it to "write access". should be in line with Github terminoloogy